Source/UZKArchive.m

```diff
@@ -2652,7 +2683,7 @@ + (zip_fileinfo)zipFileInfoForDate:(NSDate *)fileDate
     zi.tmz_date.tm_mon = (uInt)date.month - 1; // 0-indexed
     zi.tmz_date.tm_year = (uInt)date.year;
     zi.internal_fa = 0;
-    zi.external_fa = 0;
+    zi.external_fa = 2175008768;                // default value: 644
```

Comment:
I'm not sure we should be defining a nonzero default here. What's the reason for it? Also, why use the decimal form instead of an octal like `0644U`? Then you wouldn't need the comment.

Reply:
Since the original default value generated by the zip package causes the unzipped file permissions to be raised to 755, this will cause a permission attack. So I modified the default value to lower the expectation（Sorry, my English is very bad.）

Reply:
I think 755 is fine as a default - universal permissions for owner, and read/exec for everyone else. I'm not sure what you mean by "attack", but all it means is that if the developer using UnzipKit doesn't specify permissions, the permissions will be pretty wide open on the extracted file. If someone is working on security-critical archives, it's going to be up to the developer to specify UnzipKit permissions. (Also, no need to apologize for your English - it's much better than my Chinese :-)

Reply:
A [comment I made below|https://github.com/abbeycode/UnzipKit/pull/84/files/7db8290a1129b75be39afddb055c1ae0303d2afa#r290028686} pertains to this thread.

Reply:
Under the default 755 permissions, some malicious programs will be able to obtain high-privilege access to the zip and malicious attacks. I think 644 is enough for normal programs to execute.
Follow-up here can do if else to try